04-onion-routing.md

```diff
@@ -635,6 +637,35 @@ Alternatively, implementations may choose to apply non-strict forwarding only to
 like-policy channels to ensure their expected fee revenue does not deviate by
 using an alternate channel.
 
+## Resource Bucketing
+
+When making the decision to forward a payment on its outgoing channel, a node 
+MAY choose to limit its exposure to HTLCs that put it at risk of a denial of 
+service attack.
+* `unknown_allocation_slots`: defines the number of HTLC slots allocated to 
+   unknown traffic (default: 50% of remote peer's `max_accepted_htlcs`).
+* `unknown_allocation_liquidity`: defines the amount of the channel balance 
+   that is allocated to unknown traffic (default: 50% of remote peer's 
+   `max_htlc_value_in_flight_msat`.
+
+A node implementing resource bucketing limits exposure on its outgoing channel:
+- MUST choose `unknown_allocation_slots` <= the remote channel peer's 
+  `max_accepted_htlcs`.
+- MUST choose `unknown_allocation_liquidity` <= the remote channel peer's 
+  `max_htlc_value_in_flight_msat`.
+- If `endorsed` is set to 1 in the incoming `update_add_htlc` AND the HTLC 
```

Comment:
Is there a particular reason the endorsement is defined as a binary value, rather than a real number [0, 1], which may be more aptly called a `reputation_weight` or `confidence_score`?

While node’s may choose to use simpler implementations to begin (i.e binary signals and bucketed HTLCs), encoding more information in this field would allow for more precise reputation algorithms to develop independently over time without requiring a protocol change. For example, consider a long path constructed by a malicious actor:

A -> B -> … -> Y -> Z

Using only a binary signal, HTLC endorsement might get propagated through every hop. But if this is any real number, the confidence score can naturally decline and the attacker stands to lose the most reputation.